### ISSUE TITLE
Missing the option to add post build handler

### ISSUE THREAD

**Issue by lisamariet** [AutoFixture/AutoFixture#1258]
I can't find a way to specify a post build action on an object... I have tried this:

Added Customization:
`
fixture.Customize<CosmosEntity>(c => c
                .Without(p => p.CosmosId)
                .Do(a =>
                {
                    a.CosmosId = $"{a.ClientId}:{a.Key}";
                }));
`
(This did not work at all.. CosmosId is set to normal generated value, so I wonder if the code inside Do is not run?? )

or

Added Do action on Build:
`
var userSetting = _fixture.Build<CosmosEntity>()
                .Without(p => p.CosmosId)
                .Do(o =>
                {
                    o.CosmosId = $"{o.ClientId}:{o.Key}";
                })
                .Create();
`

Basically am configuring a spesific pattern for cosmosId to be as values from other properties on the object. The above code does not generate the correct CosmosId value. The first seems to have no effect with the Customize. The second the other properties are NULL at the time the Dd() runs so the value for CosmosId is set to ":".

I can off course do it after the object is created by AutoFixture but I would be really nice to not have to specify this in every new creation in every test. 
`userSetting.CosmosId = $"{userSetting.ClientId}:{userSetting.Key}";`

Many thanks.

**Comment by ecampidoglio** [comment]
Unfortunately, what you're trying to do isn't supported in AutoFixture.

The `Do` method executes the `Action<T>` delegate _during_ the creation of the specimen, not after. This means that you can't make assumptions about the state of the object that gets passed to the function, since AutoFixture may still populate some of the object's properties later on. Which properties are set to what values depends, of course, on the customizations that are applied.

`Do` is meant to be used in scenarios where you want to set the specimen's state based on something external to AutoFixture, or [invoke a method on the specimen itself](https://blog.ploeh.dk/2009/06/09/CallingMethodsWhileBuildingAnonymousVariablesWithAutoFixture/).

The workaround, as you mentioned, is to manually set the `CosmosId` property after the specimen is created.

**Comment by ecampidoglio** [comment]
Duplicate of #988
